### PR TITLE
style(Modal): fix modal content overflow issue

### PR DESCRIPTION
**Pull request by pudek357**
[SPT-1708](https://toptal-core.atlassian.net/browse/SPT-1708)
[SPT-1712](https://toptal-core.atlassian.net/browse/SPT-1712)

### Description
As `modalContent` is the last element before `children` we should add padding to this element instead of `wrapper`
https://github.com/toptal/picasso/blob/1be0c8d2075a3b7cae6f3681da85d05e93e60f37/packages/picasso/src/ModalContent/ModalContent.tsx#L34-L43
It will solve following overflow issues:
- SPT-1708: extra scrollbar for negative margins within `children`
- SPT-1712: clipped boxShadow while focusing input

### How to test
1. [https://picasso.toptal.net/SPT-1712-SPT-1708-fix-modal-content-overflow-issue/](https://picasso.toptal.net/SPT-1712-SPT-1708-fix-modal-content-overflow-issue/?9f6a73d189c89cae856be7ffddfe92136e91b40b)
2. Go to `Modal & PromptModal` stories.
3. check all examples

### Screenshots SPT-1708 (extra scrollbar for negative margins within `children`)
| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
|  |  |

### Screenshots SPT-1712 (clipped boxShadow while focusing input)
| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
|  |  |

### Review
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/api-principles.md)
- [x] Annotate all `props` in component with documentation
- [x] Create `examples` for component
- [x] Ensure that deployed demo has expected results and good examples
- [x] Ensure that tests pass by running `yarn test`
- [x] Ensure that visuals tests pass by running `yarn test:visual`. If not - check the documentation [how to fix visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md#fixing-broken-visual-tests-inside-a-pr)
- [x] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

### CODE REVIEW

**Comment by pudek357** [comment]
@toptal-bot run visual

File: packages/picasso/src/ModalContent/styles.ts [hunk 1]
```diff
@@ -1,37 +1,78 @@
 import { createStyles } from '@material-ui/core'
 
 const WRAPPER_PADDING = '2em'
+const SHADE_HEIGHT = '80px'
+const BACKGROUND_STARTING_COLOR = '#fff'
+const BACKGROUND_FINISHING_COLOR = 'rgba(255, 255, 255, 0)'
```
Comment (dmaklygin1) [inline packages/picasso/src/ModalContent/styles.ts:6]: shouldn't we reuse an existing color from the library?

File: packages/picasso/src/ModalContent/styles.ts [hunk 2]
```diff
@@ -1,37 +1,81 @@
...
+      bottom: 0,
+      '&:before': {
+        ...shadePseudoBeforeStyles,
+        backgroundColor: BACKGROUND_STARTING_COLOR,
+        bottom: 0
+      },
+      '&:after': {
+        ...shadePseudoAfterStyles,
+        bottom: WRAPPER_PADDING,
+        background: `linear-gradient(0deg, ${BACKGROUND_STARTING_COLOR} 0%, ${BACKGROUND_FINISHING_COLOR} 5rem)`
```
Comment (dmaklygin1) [inline packages/picasso/src/ModalContent/styles.ts:77]: wdyt if we have only one pseudo element with such style:
```
height: calc(2rem + 80px);
background: linear-gradient(0deg, #fff 0%, #fff 2rem, rgba(0, 0, 0, 0) 5rem);
```

Comment (dmaklygin1) [reply packages/picasso/src/ModalContent/styles.ts:77]: if yes, it means, we do not need to use pseudo at all, don't we?

Comment (pudek357) [reply packages/picasso/src/ModalContent/styles.ts]: Yeah, it's possible to do it with only one pseudo-element, I've pushed changes.

But we need this one pseudo-element as linear background and full color background has different purposes, please see video:
https://user-images.githubusercontent.com/4816942/122399773-4dfe5880-cfa5-11eb-85c2-406bc606202d.mp4

File: packages/picasso/src/ModalContent/styles.ts [hunk 3]
```diff
@@ -1,37 +1,42 @@
...
+  right: WRAPPER_PADDING,
+  left: WRAPPER_PADDING,
+  height: `calc(${SHADE_HEIGHT} + ${WRAPPER_PADDING})`
 }
 
-export default () =>
-  createStyles({
+export default ({ palette }: Theme) => {
+  const BACKGROUND_STARTING_COLOR = palette.background.default
+  const BACKGROUND_FINISHING_COLOR = hexToRgb(`${palette.common.black}0`)
```
Comment (dmaklygin1) [inline packages/picasso/src/ModalContent/styles.ts:17]: actually, the finishing color is completely transparent color. So, I would not use such construction.

**Review by dmaklygin1 — APPROVED** [review]
The rest is fine for me. Good work :)

**Review by hweeTan — APPROVED** [review]
Tested on temploy. Works as expected 💯 
image

**Pull request merged**